feat: add implementation of `stats/base/dists/hypergeometric/mode`

manvith2003 opened the pull request:
Resolves #3663

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/hypergeometric/mode along with relevant tests, benchmarks and examples

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves [[RFC]: Add C implementation for @stdlib/stats/base/dists/hypergeometric/mode #3663](https://github.com/stdlib-js/stdlib/issues/3663)

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

manvith2003 committed:
feat: add implementation of stats/base/dists/hypergeometric/mode

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

stdlib-bot committed:
chore: update copyright years

Planeshifter committed:
Merge branch 'develop' into hypergeometric_mode

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (changes requested):
This PR should be refactored to have the parameters of the distribution be typed as `int_32` instead of `double`; see other packages in https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/base/dists/hypergeometric for reference.

Planeshifter committed:
refactor: use int32_t instead of double for hypergeometric distribution parameters

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
fix: update native tests to work with int32_t parameters

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter committed:
chore: remove unneeded cppcheck directive

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter committed:
docs: fix section tags

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/hypergeometric/mode/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	for ( i = 0; i < 100; i++ ) {
+		N[i] = random_int( 1, 100 );
+		K[i] = random_int( 0, N[i] );
+		n[i] = random_int( 0, N[i] );
```
```suggestion
		N[ i ] = random_int( 1, 100 );
		K[ i ] = random_int( 0, N[ i ] );
		n[ i ] = random_int( 0, N[ i ] );
```

Planeshifter committed:
style: add spaces

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/hypergeometric/mode/lib/native.js:
```diff
@@ -0,0 +1,80 @@
+* var v = mode( 10, 5, 12 );
+* // returns NaN
+*
+* @example
+* var v = mode( 10.3, 10, 4 );
+* // returns NaN
+*
+* @example
+* var v = mode( 10, 5.5, 4 );
+* // returns NaN
+*
+* @example
+* var v = mode( 10, 5, 4.5 );
+* // returns NaN
+*
+* @example
+* var v = mode( NaN, 10, 4 );
+* // returns NaN
+*
+* @example
+* var v = mode( 20, NaN, 4 );
+* // returns NaN
+*
+* @example
+* var v = mode( 20, 10, NaN );
+* // returns NaN
```
```suggestion
* @example
* var v = mode( 10, -2, 4 );
* // returns NaN
```

Planeshifter committed:
docs: update examples

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Should be fine to land now.
